packages/tableofcontents.lua

```diff
@@ -23,9 +24,15 @@ local writeToc = function ()
   if not tocfile then return SU.error(err) end
   tocfile:write("return " .. tocdata)
   tocfile:close()
+
+  if not pl.tablex.deepcompare(SILE.scratch.tableofcontents, _tableofcontents) then
+    io.stderr:write("\n! Warning: table of contents has changed, please rerun SILE to update it.")
```

Comment:
I think this was a missed opportunity. It could have been written such that SILE would return a non-zero return value.

Perhaps… :monocle_face: 

```
[fred@🍇葡萄🍇~]$ rg '#define\s+EAGAIN' /usr/include/ 
/usr/include/asm-generic/errno-base.h
15:#define	EAGAIN		11	/* Try again */
```

Reply:
@ctrlcctrlv See https://github.com/sile-typesetter/sile/issues/230#issuecomment-916789361 ;)